File: src/main/java/com/booking/movieticket/entity/Cinema.java
package com.booking.movieticket.entity;

import com.booking.movieticket.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "cinemas")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Cinema extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequence_cinema")
    @SequenceGenerator(name = "sequence_cinema")
    @Column(name = "cinema_id")
    private Long id;

    @Column(name = "cinema_name", nullable = false)
    private String name;

    @Column(name = "hotline", nullable = false)
    private String hotline;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "logo_url", columnDefinition = "VARCHAR(1000)")
    private String logoUrl;

    @Column(name = "address")
    private String address;

    @OneToMany(mappedBy = "cinema", cascade = CascadeType.ALL)
    private Set<Branch> branches = new HashSet<>();
}

File: admin-interface/src/types/cinema.ts
export interface Cinema {
  id: number;
  name: string;
  hotline: string;
  description: string;
  logoUrl: string;
  address: string;
  isDeleted: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CinemaFormData {
  id?: number;
  name: string;
  hotline: string;
  description: string;
  logoUrl?: File;
  address: string;
} 